### ISSUE TITLE
TexStudio has crashed due to a SIGSEGV after Ubuntu updating

### ISSUE THREAD

**Issue by vlamor** [texstudio-org/texstudio#825]
### Environment
- TeXstudio: 2.12.16 
- Qt: 5.9.5
- OS: Ubuntu 18.04.3 LTS
- TeX distribution: texlive

After software updating in Ubuntu which made automatically TexStudio began crash after compiling of any (even empty) .tex file.

The texstudio_backtrace1.txt file is as following:

SIGSEGV (TXS-Version  R )
[bt] /usr/bin/texstudio(_ZN12SimulatedCPU13set_from_realEv+0) [0x55b34b74aea0]

**Comment by MeanSquaredError** [comment]
This does not look like a complete stack trace. Can you please confirm that it is just 2 lines that you pasted?

If it is indeed just two lines, then can you please get a core dump for the crashing TeXstudio? I am not an Ubuntu user but it seems that on Ubuntu you would need to enable core dumps from the command line:

`ulimit -c unlimited`
`sudo sysctl -w kernel.core_pattern=core.%u.%p.%t`
`/usr/bin/texstudio`
Then make TeXstudio crash and check for a `core....` file in the current directory.

The last line starts TeXstudio so you may need to adjust it to point to the actual location of TeXstudio in your filesystem.
For details about enabling crash dumps on Ubuntu you can see
https://askubuntu.com/questions/966407/where-do-i-find-the-core-dump-in-ubuntu-16-04lts
(but it seems that enabling Apport is not necessary if you just enable the coredump by changing core_pattern as in the example above.

**Comment by vlamor** [comment]
Thank you for your reply. I've run commands:
````
ulimit -c unlimited
sudo sysctl -w kernel.core_pattern=core.%u.%p.%t
/usr/bin/texstudio
````
and received in termianl
crashed with signal SIGSEGV
SIGSEGV (TXS-Version  R )
SIGSEGV (TXS-Version  R )
SIGSEGV (TXS-Version  R )
SIGSEGV (TXS-Version  R )

[bt] /usr/bin/texstudio(_ZN12SimulatedCPU13set_from_realEv+0) [0x555aabf11ea0]
[bt] /usr/bin/texstudio(_ZN12SimulatedCPU13set_from_realEv+0) [0x555aabf11ea0]

"/tmp/texstudio_backtrace1.txt"
"Current: 0/0"
Killed on user request after error: SIGSEGV

Aborted (core dumped)

And received 195 MB of binary file with core dump
[core-1000.8753.1575631180.zip](https://github.com/texstudio-org/texstudio/files/3931932/core-1000.8753.1575631180.zip)

**Comment by MeanSquaredError** [comment]
Hi,
Thank you for the core dump file. Can you please also zip and upload the binary for TeXstudio that is installed on your system? I expect that it must be located at /usr/bin/texstudio but the exact location might differ on Ubuntu.

**Comment by vlamor** [comment]
Hi,

Thank you for your help. Please find as attached .zip binary
[texstudio.zip](https://github.com/texstudio-org/texstudio/files/3934995/texstudio.zip)
 texstudio file

**Comment by MeanSquaredError** [comment]
Thank you for the information. I examined the core file together with the texstudio binary that you provided. Here is what I have found:

1. The core dump does not contain any program header sections, so nether the standard debugger (gdb) nor any custom program could fetch a meaningful backtrace. So at this point the provided core dump is not useful.
2. The binary for texstudio that you provided runs correctly on my machine (64-bit Fedora). It also compiles correctly .tex files.
3. When I try killing texstudio with SIGSEGV (the system signal that killed TeXstudio on your computer) it also provides a useless 1-line backtrace pretty much like the one that you sent. So there is some kind of bug in the TeXstudio crash handling code that needs to be investigated. Your executable file of texstudio dates back to Oct 05, so the bug has been there at least for a couple of months. However I do not think that the bug in the crash-handling code is really what causes TeXstudio to crash on your computer.

So at this point I would say that the easiest thing to try is to use the Ubuntu's built-in apport-retrace to get a meaningful crash backtrace. Please do the following:

1. Make sure that the computer has been rebooted at least once since the last time when you did
sudo sysctl -w kernel.core_pattern=core.%u.%p.%t
Basically the reboot makes sure that the changes to kernel.core_pattern have been cleared and apport gets control on program crash.

2. Install the debug symbols for texstudio. I think the corresponding package is called "texstudio-dbgsym"
3. Run texstudio and make it crash.
4. When apport shows the popup after the crash click on "Examine locally", select "Run gdb session" and then a terminal with gdb session will appear.
5. Save and paste here the part of the gdb session with the backtrace. That is the final part starting with the line "Program terminated with signal 11..."

You can see the process of getting an apport backtrace explained here:
https://wiki.ubuntu.com/DebuggingProgramCrash
(they provide helpful screenshots too).

**Comment by sunderme** [comment]
is it ongoing ?

**Comment by MeanSquaredError** [comment]
@sunderme 

I tested the texstudio binary that the user provided and it works correctly including correct compilation of .tex files to .pdf

So my guess is that something went wrong during Ubuntu's upgrade and some of the system libraries are damaged which in turn causes TXS to crash.

In order to prove this suspicion we would need a proper stack trace trace and/or a proper crash dump. The original stack trace is incorrect and the core dump lacks a program header section so it is not possible to get the correct EIP (program counter) for any thread.

If the user follows the 5 steps that I described above and gets a proper stack trace through Ubuntu's apport, then we can investigate the issue. Until then not much can be done.
